### PR TITLE
Lib1 performance comparison tests with different codecs

### PR DESCRIPTION
**Pull request by fs3itz**
```
Codec: ffv1
Loading 10 Files took 1241 ms (124 ms / frame)
File size reduced from 2.3 MB to 1.9 MB (83.3%)

Codec: ffv1
Loading 100 Files took 6059 ms (60 ms / frame)
File size reduced from 23.6 MB to 18.3 MB (77.6%)

Codec: ffv1
Loading 500 Files took 29272 ms (58 ms / frame)
File size reduced from 113.2 MB to 86.1 MB (76.1%)

Codec: ffv1
Loading 800 Files took 45525 ms (56 ms / frame)
File size reduced from 173.3 MB to 133.4 MB (77.0%)

Codec: ffv1
Loading 1000 Files took 58758 ms (58 ms / frame)
File size reduced from 220.6 MB to 169.7 MB (76.9%)


Codec: vp9
Loading 10 Files took 1804 ms (180 ms / frame)
File size reduced from 2.7 MB to 1.5 MB (54.2%)

Codec: vp9
Loading 100 Files took 17195 ms (171 ms / frame)
File size reduced from 24 MB to 13.7 MB (57.1%)

Codec: vp9
Loading 500 Files took 88192 ms (176 ms / frame)
File size reduced from 114.6 MB to 63 MB (55.0%)

Codec: vp9
Loading 800 Files took 138997 ms (173 ms / frame)
File size reduced from 179.1 MB to 102.6 MB (57.3%)

Codec: vp9
Loading 1000 Files took 164804 ms (164 ms / frame)
File size reduced from 225.4 MB to 129.7 MB (57.6%)
```

### CODE REVIEW

**Review by simonsasse — CHANGES_REQUESTED** [review]
looks good, except pls apply ktlint. I know that there are ktlint errors in other parts of the project, but I think it's good to try to not add more code that makes ktlint go off. Interesting results, thx ! :)

File: VideoGenerator/library/src/main/kotlin/VideoGeneratorStressTest.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.io.File
+import java.lang.Exception
+import java.text.DecimalFormat
+import kotlin.math.log10
+import kotlin.math.pow
+import kotlin.system.measureTimeMillis
+
+class VideoGeneratorStressTest {
```
Comment (akriese) [inline VideoGenerator/library/src/main/kotlin/VideoGeneratorStressTest.kt:13]: This is probably more of a benchmark than a stress test, technically :)

File: VideoGenerator/library/src/main/kotlin/VideoGeneratorStressTest.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    private lateinit var videoGenerator: VideoGeneratorImpl
+    private val videoPath = "src/resources/testOutput.mkv"
+    private val inputPath = "../example/app/src/androidTest/assets/screen"
+    private val testCodecs =
+        mapOf<Int, Map<String, String>?>(
```
Comment (akriese) [inline VideoGenerator/library/src/main/kotlin/VideoGeneratorStressTest.kt:18]: Why are the other codecs commented out? did they not conform to our requirements?

Comment (fs3itz) [reply VideoGenerator/library/src/main/kotlin/VideoGeneratorStressTest.kt:18]: yes either they did not conform our licence requirements or somehow did not produce a lossless output
JPEG2000 was significantly worse so i commented it out as well
YUV4 was just to compare to raw data

**Comment by akriese** [comment]
Getting similar results, looks fine to me

**Comment by akriese** [comment]
Please add an info to `VideoGenerator` on how to run the tests in `library/src/main/kotlin` from the command line, I only did it in IntelliJ

**Review by akriese — APPROVED** [review]

**Pull request merged**